src/flags/index.js

```diff
@@ -20,7 +20,7 @@ export type Flag <T> = {
   completion?: ?{
     cacheDuration?: ?number,
     cacheKey?: ?string,
-    options: () => Promise<string[]>
+    options: (?any) => Promise<string[]>
```

Comment:
in other words, cli-engine should always be passing in the same objects to the cache function otherwise you'd have to have logic for every flag manually configured in cli-engine